### PR TITLE
[Merged by Bors] - fix(GetAllModules): sort module names instead of file names

### PR DESCRIPTION
**Pull request by grunweg**
Different platforms have different path separators, making the sorting of files by their names platform-dependent.
This meant the output `mk_all` would depend on which platform it was run on.
To avoid this, we sort the resulting module names instead --- for which the path separators have been converted to dots.

To make this difference clear, mention this in the documentation and rename to `getAllModulesSorted`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Util/GetAllModules.lean [hunk 1]
```diff
@@ -39,18 +39,18 @@ def getAllFiles (git : Bool) (ml : String) : IO (Array System.FilePath) := do
     else do
       let all ← walkDir ml
       return all.filter (·.extension == some "lean"))
-  let files := (allModules.erase ml.lean).qsort (·.toString < ·.toString)
-  let existingFiles ← files.mapM fun f => do
+  let existingFiles ← (allModules.erase ml.lean).mapM fun f => do
```
Comment (adomani) [inline Mathlib/Util/GetAllModules.lean:42]: Does `filterMapM` exist?  And can it be used here?

**Review by adomani — APPROVED** [review]

**Comment by adomani** [comment]
Thanks for this: looks good to me!

I'll wait to see if there are more comments, but I think that this is an improvement!

File: Mathlib/Util/GetAllModules.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Util/GetAllModules.lean:42]: It exists, TIL. I'll take a look!

Comment (grunweg) [reply Mathlib/Util/GetAllModules.lean]: Used it here, thanks for the pointer!

**Comment by adomani** [comment]
I'll just

maintainer merge

this, thanks!

**Comment by adomani** [comment]
maintainer merge

**Comment by bryangingechen** [comment]
Thanks!
bors r+

**Pull request closed without merging**